src/components/ActivityFeed/ActivityTreasury.tsx

```
@@ -0,0 +1,464 @@
+			const treasuryBalance = { freeBalance: freeBalance as Balance };
+
+			updateBurnValue(treasuryBalance, currentTokenPrice, setNextBurn);
+			updateAvailableValue(treasuryBalance, currentTokenPrice, network, setAvailable);
+		} catch (error) {
+			console.error(error);
+			setAvailable({ isLoading: false, value: '', valueUSD: '' });
+			setNextBurn({ isLoading: false, value: '', valueUSD: '' });
+		}
+	};
+
+	const updateBurnValue = (treasuryBalance: any, currentTokenPrice: TokenPrice, setNextBurn: Function) => {
```

Biome

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)